modules/renter/contractor/journal.go

```diff
@@ -0,0 +1,332 @@
+package contractor
+
+// The contractor achieves efficient persistence using a JSON transaction
+// journal. It enables efficient ACID transactions on JSON objects.
+//
+// The journal represents a single JSON object, containing all of the
+// contractor's persisted data. The object is serialized as an "initial
+// object" followed by a series of update sets, one per line. Each update
```

Comment:
Another thing that would greatly improve readability would be to use `json:",omitempty"` on all our fields. In my tests this seemed to shrink the journal by 10-15%. But there are API compatibility concerns: for anything exposed to the API, we would need to preserve the current behavior.